services/spark-operator/1.1.6/defaults/cm.yaml

```diff
@@ -5,4 +5,17 @@ metadata:
   name: spark-operator-1.1.6-d2iq-defaults
   namespace: ${releaseNamespace}
 data:
-  values.yaml: ""
+  values.yaml: |-
+    historyServer:
+      enabled: true
```

Comment:
I wanted to add a clarification here.

It is safe to set `historyServer.enabled=true` by itself, however this option is used in conjunction with `historyServer.fs.logDirectory` set to S3, HDFS or some shared volume mount.  This is described in this [comment](https://github.com/mesosphere/spark-on-k8s-operator/blob/d2iq-spark-operator-chart-1.1.6/charts/spark-operator-chart/values.yaml#L182-L184)

With `historyServer.enabled=true` && `historyServer.fs.logDirectory=""` Spark will start but the history-server metadata will land within the pod and won't be accessible externally, this is the failsafe setting.

I suggest adding `historyServer.fs.logDirectory` as a commented out option to go along with `historyServer.enabled` so its very clear to the user that these two settings go together. 

I think also we'll need to a docs section similar to [KUDO Spark Operator](https://kudo.dev/docs/runbooks/spark/history-server.html#installing-spark-operator-with-history-server-enabled) for this and other part of Spark.

Reply:
So the default values are meant to be configuration with which a deployment works out of the box. If setting `historyServer.enabled` to `true` requires another setting to be set in order for things to work properly, then I think we need to disable that option in the defaults. If the user wants to turn it on, then they would need to add their custom config on top to override our defaults.

> With historyServer.enabled=true && historyServer.fs.logDirectory="" Spark will start but the history-server metadata will land within the pod and won't be accessible externally, this is the failsafe setting.

This sounds like maybe we can go with this, as long it isn't preventing spark from running, or generating a bunch of errors or failed pods that would alarm a user? Once a user sets that config, would things resolve itself or would there be a manual step to restart something? Either way, we would def need to add docs on the app description page that the user should set the `historyServer.fs.logDirectory` setting in order to actually use the feature

Reply:
>This sounds like maybe we can go with this, as long it isn't preventing spark from running, or generating a bunch of errors or failed pods that would alarm a user? Once a user sets that config, would things resolve itself or would there be a manual step to restart something?

It won't prevent spark from running, just that it might be confusing as there's two options that need to be configured simultaneously for it to be useful. Once a useful value for `historyServer.fs.logDirectory` is set, you'll need a restart of the pod for the data to appear.

I think let's set `historyServer.enabled=false` even if its safe to have it on by default, this should indicate to the user that an option for History Server exists but that additional accompanying settings are needed to enable it correctly.

We can link to the docs about the accompanying `historyServer.fs.logDirectory` setting. How does that sound?